resources/tentative/complex-static-html/src/big-dom-generator/gen-css.js

```diff
@@ -159,80 +121,108 @@ const randomWeighted = (options, probs) => {
+                }`;
+    });
 };
 
 const cssProperties = ["accent-color", "border-bottom-color", "border-color", "border-left-color", "border-right-color", "border-top-color", "column-rule-color", "outline-color", "text-decoration-color"];
 
-// Returns a random 200 matching selectors and 200 non-matching selectors targeted at the todoMVC items.
-export const genCss = () => {
+/**
+ * Generates CSS for the matching and non-matching selectors.
+ * @param {string} angular whether to generate angular or react markup
+ * @returns {string} The css rules for the matching and non-matching selectors.
+ */
+export const genCss = (angular = false) => {
     const matchingSelectors = [];
     const nonMatchingSelectors = [];
-    for (let index = 0; index < 100; index++) {
-        // Add `.targeted` to the matching selectors to match only the todoMVC items.
-        matchingSelectors.push(`${buildMatchingSelector(6, index, "", 0, random.randRange(3, MAX_SELECTOR_LENGTH_TO_GENERATE))}.targeted`);
-        matchingSelectors.push(`${buildMatchingSelector(7, index, "", 0, random.randRange(3, MAX_SELECTOR_LENGTH_TO_GENERATE))}.targeted`);
-        nonMatchingSelectors.push(buildNonMatchingSelector(6, index, "", 0, random.randRange(3, MAX_SELECTOR_LENGTH_TO_GENERATE)));
-        nonMatchingSelectors.push(buildNonMatchingSelector(7, index, "", 0, random.randRange(3, MAX_SELECTOR_LENGTH_TO_GENERATE)));
-    }
-    // Create random color styles. Same color, different opacity.
-    // TODO: Choose a better color for the todoMVC theme.
-    const matchingCssRules = [];
-    matchingSelectors.forEach((selector, i) => {
-        random.shuffle(cssProperties, true);
-        matchingCssRules.push(`${selector} { 
-            ${cssProperties[0]}: rgba(140,140,140,${i / 1000}); 
-            ${cssProperties[1]}: rgba(140,140,140,${i / 1000});
-        }`);
-    });
-    const nonMatchingCssRules = [];
-    nonMatchingSelectors.forEach((selector, i) => {
-        random.shuffle(cssProperties, true);
-        nonMatchingCssRules.push(`${selector} { 
-            ${cssProperties[0]}: rgba(140,140,140,${i / 1000}); 
-            ${cssProperties[1]}: rgba(140,140,140,${i / 1000});
-        }`);
+    const htmlMarkup = getHtmlMarkup(angular);
+    const dom = new JSDOM(htmlMarkup);
+    const { document } = dom.window;
+
+    addTodoItems(document, NUM_TODOS_TO_INSERT_IN_HTML, angular);
+    const elements = document.querySelectorAll(".main li");
+
+    // Generate matching and non-matching selectors for each element.
+    elements.forEach((element) => {
+        const chooseFrom = [element, element.firstChild];
+        random.shuffle(chooseFrom, true);
+        // Add `TARGETED_CLASS` to the matching selectors to match only the todoMVC items.
+        matchingSelectors.push(`${buildMatchingSelector(chooseFrom[0], getInitialDepth(chooseFrom[0], angular), "", 0, random.randRange(3, MAX_SELECTOR_LENGTH_TO_GENERATE))}${TARGETED_CLASS}`);
```

Comment:
No need for the `chooseFrom` array. Use `element` and `element.firstChild` directly